pkg/util/limiter/query_limiter.go

```diff
@@ -75,7 +80,7 @@ func QueryLimiterFromContextWithFallback(ctx context.Context) *QueryLimiter {
 	ql, ok := ctx.Value(ctxKey).(*QueryLimiter)
 	if !ok {
 		// If there's no limiter return a new unlimited limiter as a fallback
-		ql = NewQueryLimiter(0, 0, 0)
+		ql = NewQueryLimiter(0, 0, 0, nil)
```

Comment:
This means the metrics struct may be nil, but the rest of the code doesn't handle that case.

Reply:
In the case where the limits are 0, the checks are skipped and the metrics struct isn't used.